carmm/analyse/bonds.py

```diff
@@ -253,3 +253,80 @@ def print_bond_table_header():
     print('{:<6.5s}{:<6.5s}{:>4.10s}{:^13.10s}{:>4.10s}'.format(
         "Bond", "Count", "Average", "Minimum", "Maximum"))
     print("-" * 40)
+
+def analyse_chelation(atoms, metal, ligand_atom, mult=1):
+    '''
+    Returns information on the ligands surrounding a metal atom and their chelation type. Currently only works with one metal atom.
+    TODO: rework so script can account for multiple separate atoms.
+
+    Parameters:
+    atoms: Atoms object
+        Input structure from which to calculate molecular information
+    Mult: float value
+        Multiplier for the bond cutoffs. Set to 1 as default but can be adjusted depending on application
+    metal: String
+        Metal atom to characterise the coordination environment around
+    ligand_atom: String
+        Element symbol of the atom coordinating with the metal atom
+        TODO: expand this functionality as a list of element symbols.
+    '''
+
+    ## Import modules
+    from carmm.analyse.bonds import analyse_bonds, analyse_all_bonds
+    from ase.neighborlist import NeighborList, get_connectivity_matrix, natural_cutoffs
+    from scipy import sparse
+    import collections
+    from collections import OrderedDict
+
+    ## identifies atoms coordinated to the metal cation
+    ligand_coord = analyse_bonds(atoms, metal, ligand_atom, verbose=False)
+
+    ## Defines cutoffs, connectivity matrix for first image.
+    cutOff = natural_cutoffs(atoms, mult=mult)
+    neighborList = NeighborList(cutOff, skin=0, self_interaction=True, bothways=True)
+    neighborList.update(atoms)
+    # defines matrix and removes entries from the metal, so the complex is not considered as one molecule.
+    matrix = neighborList.get_connectivity_matrix(sparse=False)
```

Comment:
Would probably rename this variable so it's a little clearer what it does further in the code.